radixdlt-java/src/main/java/com/radixdlt/client/core/atoms/Atom.java

```diff
@@ -197,20 +201,53 @@ public EUID getHid() {
 			.collect(Collectors.toList());
 	}
 
-	public Map<TokenClassReference, Map<ECPublicKey, BigInteger>> tokenSummary() {
-		return this.ownedTokensParticles()
+	public Map<TokenTypeReference, Map<ECPublicKey, BigInteger>> tokenSummary() {
+		Map<TokenTypeReference, Map<ECPublicKey, BigInteger>> consumableTokens = this.consumableTokens()
+			.collect(Collectors.groupingBy(
+				tokens -> tokens.getFirst().getTokenTypeReference(),
+				Collectors.groupingBy(
+					tokens -> tokens.getFirst().getOwner(),
+					Collectors.reducing(BigInteger.ZERO, this::consumableToAmount, BigInteger::add)
+				)
+			));
+
+		Map<TokenTypeReference, Map<ECPublicKey, BigInteger>> consumingTokens = this.consumingTokens()
 			.collect(Collectors.groupingBy(
-				s -> s.getParticle().getTokenClassReference(),
+				tokens -> tokens.getFirst().getTokenTypeReference(),
 				Collectors.groupingBy(
-					s -> s.getParticle().getOwner(),
-					Collectors.reducing(BigInteger.ZERO, Atom::ownedTokensToBigInteger, BigInteger::add)
+					tokens -> tokens.getFirst().getOwner(),
+					Collectors.reducing(BigInteger.ZERO, this::consumingToAmount, BigInteger::add)
 				)
 			));
+
+		return Stream.of(consumableTokens, consumingTokens)
+			.flatMap(map -> map.entrySet().stream())
+			.collect(
+				Collectors.toMap(
+					Map.Entry::getKey,
+					Map.Entry::getValue,
+					(consumables, consuming) ->
+						Stream.of(consumables, consuming)
+							.flatMap(map -> map.entrySet().stream())
+							.collect(
+								Collectors.toMap(
+									Map.Entry::getKey,
+									Map.Entry::getValue,
+									BigInteger::add
+								)
+							)
+				)
+			);
+	}
+
+	private BigInteger consumableToAmount(Pair<ConsumableTokens, Spin> tokens) {
+		BigInteger amount = UInt256s.toBigInteger(tokens.getFirst().getAmount());
+		return tokens.getSecond() == Spin.DOWN ? amount.negate() : amount;
 	}
 
-	private static BigInteger ownedTokensToBigInteger(SpunParticle<OwnedTokensParticle> value) {
-		BigInteger bi = UInt256s.toBigInteger(value.getParticle().getAmount());
-		return (value.getSpin() == Spin.UP) ? bi : bi.negate();
+	private BigInteger consumingToAmount(Pair<ConsumingTokens, Spin> tokens) {
+		BigInteger amount = UInt256s.toBigInteger(tokens.getFirst().getAmount());
```

Comment:
I don't think this is correct as it stands now.  There are two types of ConsumingTokens, Burned and Transferred.  Unfortunately they have different signs from each other — UP Transferred tokens mean an increase, whereas UP Burned tokens means a decrease.

Reply:
Fixed, this logic only applies to *consumables* - UP means increase, DOWN means decrease, always. I had my consumables/consuming mixed up.. the irony.